src/group_theory/perm/basic.lean

```diff
@@ -55,6 +55,33 @@ lemma eq_inv_iff_eq {f : perm α} {x y : α} : x = f⁻¹ y ↔ f x = y := f.eq_
 
 lemma inv_eq_iff_eq {f : perm α} {x y : α} : f⁻¹ x = y ↔ x = f y := f.symm_apply_eq
 
+/-! Lemmas about mixing `perm` with `equiv`. Because we have multiple ways to express
+`equiv.refl`, `equiv.symm`, and `equiv.trans`, we want simp lemmas for every combination.
+The assumption made here is that if you're using the group structure, you want to preserve it after
+simp. -/
+
+@[simp] lemma trans_one {α : Sort*} {β : Type*} (e : α ≃ β) : e.trans (1 : perm β) = e :=
+equiv.trans_refl e
+
+@[simp] lemma mul_refl (e : perm α) : e * equiv.refl α = e := equiv.trans_refl e
+
+@[simp] lemma one_symm : (1 : perm α).symm = 1 := equiv.refl_symm
+
+@[simp] lemma refl_inv : (equiv.refl α : perm α)⁻¹ = 1 := equiv.refl_symm
```

Comment:
Here and below, are we choosing that `1` is the simp normal form for a `perm` that is an `equiv.refl`? That makes sense to me if we also simp `trans` into `*`. Otherwise, there'll still be a mixture of `refl` and `trans` and `1` and `*`.

Reply:
No, we're making the weaker claim "if your expression uses any of `*`, `⁻¹`, or `1`, then the simp-normal form within that expression should use those symbols".